rdrf/rdrf/reports/generator.py

```diff
@@ -0,0 +1,707 @@
+                models.append(related_model)
+            models.extend(self._get_related_models(related_model))
+
+        return models
+
+    def _mirror_table(self, table_name, source_engine, target_engine):
+        try:
+            self._drop_table(table_name)
+        except:
+            pass
+
+        source_meta = MetaData(bind=source_engine)
+        table = alc.Table(table_name, source_meta, autoload=True)
+        table.metadata.create_all(self.reporting_engine)
+        self._copy_table_data(source_engine, target_engine, table)
+
+    def create_tables(self):
+        if self.reporting_engine is not self.default_engine:
+            logger.info("CREATING DEMOGRAPHIC TABLES")
+            self._create_demographic_tables()
+
+        logger.info("CREATING CLINICAL TABLES")
+
+        for form_model in self.registry_model.forms:
+            if form_model.name.startswith("GeneratedQuestionnaire"):
+                continue
+            logger.debug("creating table for clinical form %s" %
+                         form_model.name)
+            columns = self._create_form_columns(form_model)
+            table = self._create_table(form_model.name, columns)
+
+            single_form_table = ClinicalTable(TableType.CLINICAL_FORM,
+                                              table,
+                                              columns,
+                                              form_model)
+
+            self.clinical_tables.append(single_form_table)
+
+            for section_model in form_model.section_models:
+                if section_model.allow_multiple:
+                    logger.info("creating multisection table for %s %s" % (form_model.name,
+                                                                           section_model.code))
+
+                    columns = self._create_multisection_columns(form_model,
+                                                                section_model)
+                    table_name = form_model.name + "_" + section_model.code
+
+                    if len(table_name) > MAX_TABLE_NAME_LENGTH:
+                        table_name = table_name[:MAX_TABLE_NAME_LENGTH]
+                        
+
+                    if "!" in table_name:
+                        table_name = table_name.replace("!","")
+                        
+                    table = self._create_table(table_name, columns)
+                    multisection_table = ClinicalTable(TableType.MULTISECTION,
+                                                       table,
+                                                       columns,
+                                                       form_model,
+                                                       section_model)
+
+                    self.clinical_tables.append(multisection_table)
+
+        self._extract_clinical_data()
+
+    @property
+    def patients(self):
+        return Patient.objects.filter(rdrf_registry__in=[self.registry_model])
```

Comment:
not sure why the `__in` is needed. Can't just do `=self.registry_model`?

Reply:
it's one to many